hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -1553,9 +1553,8 @@ protected Object initializeLazyPropertiesFromDatastore(
 			partsToSelect.add( getAttributeMapping( propertyIndex ) );
 			SingleIdArrayLoadPlan lazyLoanPlan;
 			if ( nonLazyPropertyLoadPlansByName == null ) {
-				nonLazyPropertyLoadPlansByName = new HashMap<>();
+				nonLazyPropertyLoadPlansByName = new ConcurrentHashMap<>();
 				lazyLoanPlan = createLazyLoanPlan( partsToSelect );
-				;
 				nonLazyPropertyLoadPlansByName.put( fieldName, lazyLoanPlan );
 			}
 			else {
```

Comment:
```suggestion
			ConcurrentHashMap<String,SingleIdArrayLoadPlan> nonLazyPropertyLoadPlansByName = this.nonLazyPropertyLoadPlansByName;
			if ( nonLazyPropertyLoadPlansByName == null ) {
				nonLazyPropertyLoadPlansByName = new ConcurrentHashMap<>();
				lazyLoanPlan = createLazyLoanPlan( partsToSelect );
				nonLazyPropertyLoadPlansByName.put( fieldName, lazyLoanPlan );
				this.nonLazyPropertyLoadPlansByName = nonLazyPropertyLoadPlansByName;
			}
			else {
```

Also, please remove the unnecessary semicolon a few lines below.